lib/chef/knife/vsphere_vm_find.rb

```diff
@@ -0,0 +1,277 @@
+#
+# Author:: Raducu Deaconu (<rhadoo_io@yahoo.com>)
+# License:: Apache License, Version 2.0
+#
+
+require 'chef/knife'
+require 'chef/knife/base_vsphere_command'
+require 'rbvmomi'
+require 'netaddr'
+
+PsOn = 'poweredOn'
+PsOff = 'poweredOff'
+PsSuspended = 'suspended'
+
+#find vms belonging to pool that match criteria, display specified fields
+class Chef::Knife::VsphereVmFind < Chef::Knife::BaseVsphereCommand
+  banner "knife vsphere vm find"
+
+  common_options
+option :pool,
+        long: '--pool pool',
+        short: '-h',
+        description: 'Target pool'
+
+option :esx_disk,
+       long: '--esx-disk',
+       description: 'Show esx disks'
+
+option :snapshots,
+       long: '--snapshots',
+       description: 'Show snapshots'
+
+option :os_disk,
+       long: '--os-disks',
+       description: 'Show os disks'
+
+option :cpu,
+       long: '--cpu',
+       description: 'Show cpu'
+
+option :ram,
+       long: '--ram',
+       description: 'Show ram'
+
+option :ip,
+       long: '--ip',
+       description: 'Show ip'
+
+option :soff,
+       long: '--powered-off',
+       description: 'Show only stopped machines'
+       
+option :son,
+       long: '--powered-on',
+       description: 'Show only started machines'
+       
+option :matchip,
+       long: '--match-ip IP',
+       description: 'match ip'
+       
+option :matchos,
+       long: '--match-os OS',
+       description: 'match os'
+
+
```

Comment:
A minor nit, but while you're reformatting could you also get rid of any double blank lines?